src/App.tsx

```diff
@@ -1,24 +1,291 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
+import * as todoService from './api/todos';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { USER_ID } from './utils/fetchClient';
+import { Status } from './types/Status';
+import { Message } from './types/Message';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosStatus, setTodosStatus] = useState<Status>(Status.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<Message | ''>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [titleTodo, setTitleTodo] = useState('');
+  const [deletedTodo, setDeletedTodo] = useState<number[] | null>(null);
+  const [changedTodo, setChangedTodo] = useState<number[] | null>(null);
+
+  const filterTodos = (listTodos: Todo[], status: Status) => {
+    switch (status) {
+      case Status.Active:
+        return listTodos.filter(todo => !todo.completed);
+      case Status.Completed:
+        return listTodos.filter(todo => todo.completed);
+      case Status.All:
+      default:
+        return listTodos;
+    }
+  };
+
+  const addTodo = ({ userId, title, completed }: Todo) => {
+    setErrorMessage('');
+    setIsLoading(true);
+
+    const createdTodo = {
+      id: 0,
+      userId: USER_ID,
+      title: title.trim(),
+      completed: false,
+    };
+
+    setTempTodo(createdTodo);
+
+    todoService.createTodo({ userId, title, completed })
+      .then(newTodo => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+        setTitleTodo('');
+      })
+      .catch(() => {
+        setErrorMessage(Message.NoAddTodo);
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setTempTodo(null);
+      });
+  };
+
+  const removeComplatedTodos = (todosId: number[] | null) => {
+    setIsLoading(true);
+
+    if (todosId) {
+      setDeletedTodo(todosId);
+      todosId.map(todoId => {
+        return todoService.deleteTodos(todoId)
+          .then(() => {
+            setTodos(currentTodos => (
+              currentTodos.filter(todo => todo.id !== todoId)
+            ));
+          })
+          .catch((error) => {
+            setDeletedTodo(null);
+            setErrorMessage(Message.NoDeleteTodo);
+            throw error;
+          })
+          .finally(() => {
+            setIsLoading(false);
+            setDeletedTodo(null);
+          });
+      });
+    }
```

Comment:
set loading to false after this `if` block by default

Reply:
by the way, why you need this `if`? you can trigger this function only if `todosId` array is not empty